src/main/java/net/einsteinsci/betterbeginnings/register/recipe/AdvancedRecipe.java

```diff
@@ -182,19 +194,18 @@ public boolean checkMatchMostly(InventoryCrafting crafting, int width, int heigh
 
 				ItemStack craftingStackInQuestion = crafting.getStackInRowAndColumn(k, l);
 
-				if (craftingStackInQuestion != null || neededCraftingStack != null)
+				if (craftingStackInQuestion != null && neededCraftingStack != null)
 				{
-					// If one is null but not the other
-					if (craftingStackInQuestion == null || neededCraftingStack == null)
-					{
-						return false;
-					}
-
 					if (!neededCraftingStack.matches(craftingStackInQuestion))
 					{
 						return false;
 					}
 				}
+				else if ((craftingStackInQuestion == null) ^ (neededCraftingStack == null))
```

Comment:
http://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.22.2

Java documentation says this has defined behavior and works as a logical operator on boolean operands.

If you think it would be better written in a different fashion I can fix it and re-commit.